daemon/daemon.go

```diff
@@ -946,30 +935,27 @@ func NewDaemon(c *Config) (*Daemon, error) {
 			log.WithError(err).WithField(logfields.V6Prefix, v6ServicePrefix).Fatal("Invalid IPv6 allocation prefix")
 		}
 
-		if err := nodeaddress.SetIPv6NodeRange(net); err != nil {
+		if err := node.SetIPv6NodeRange(net); err != nil {
 			log.WithError(err).WithField(logfields.V6Prefix, net).Fatal("Invalid per node IPv6 allocation prefix")
 		}
 	}
 
 	if v6ServicePrefix != AutoCIDR {
-		_, _, err := net.ParseCIDR(v6ServicePrefix)
+		_, ipnet, err := net.ParseCIDR(v6ServicePrefix)
 		if err != nil {
 			log.WithError(err).WithField(logfields.V6Prefix, v6ServicePrefix).Fatal("Invalid IPv6 service prefix")
 		}
+
+		node.AddAuxPrefix(ipnet)
```

Comment:
This should be the else case for if `err != nil` since we don't exit/return on error.

Reply:
oh, nevermind, it's a fatal